pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -83,6 +94,12 @@ func (fr *FlowRecords) AddFlowRecordToMap(connKey *flowexporter.ConnectionKey, r
 	fr.recordsMap[*connKey] = *record
 }
 
+// AddFlowRecordWithoutLock adds the flow record from record map given connection key.
+// Caller is expected to grab the lock the record map.
+func (fr *FlowRecords) AddFlowRecordWithoutLock(connKey *flowexporter.ConnectionKey, record *flowexporter.FlowRecord) {
```

Comment:
maybe you can address naming consistency with the function above (`AddFlowRecordToMap`) in a separate PR, or implement the approach I suggested above if it is acceptable

Reply:
Yes this can be done. Hope you are ok with the explanation in the other comment.